programs/signer-authorization/src/lib.rs

```diff
@@ -37,7 +35,9 @@ pub mod signer_authorization {
 #[derive(Accounts)]
 pub struct InitializeVault<'info> {
     #[account(
-        init,
+        // We use init_if_needed here for the test. Otherwise, the test will result in an error "Already in use" after testing once.
```

Comment:
Are you sure? The validator started by Anchor tests doesn't keep state. Running the test suite a second time should start a new validator with blank state.

Reply:
The `solana-test-validator` command has an optional parameter:
```
-r, --reset
            Reset the ledger to genesis if it exists. By default the validator will resume an existing ledger (if
            present)
```
If you don't pass this parameter, the previous state will be retained.
I believe beginners might overlook this parameter :)